Add NuclearJS example

jordangarcia opened the pull request:
@jtremback @voronianski 

addresses #41

jordangarcia committed:
Add nuclear-js example

jordangarcia committed:
Add back lux to package.json

jordangarcia committed:
cleanup to use es6

voronianski commented on nuclear-js/js/actions.js:
```diff
@@ -0,0 +1,39 @@
+import shop from '../../common/api/shop'
```
@jordangarcia missed semicolon everywhere

voronianski commented on nuclear-js/js/actions.js:
```diff
@@ -0,0 +1,39 @@
+        reactor.dispatch(CHECKOUT_START)
+
+        shop.buyProducts(productsInCart, () => {
+            console.log("YOU BOUGHT: ", productsInCart)
```
@jordangarcia single quotes

voronianski commented on nuclear-js/js/stores/CartStore.js:
```diff
@@ -0,0 +1,53 @@
+}
+
+function finishCheckout(state) {
+    return initialState
```
@jordangarcia unused `state`, should it be returned instead of `initialState`?

jordangarcia replied:
no this should be `initialState` as its taking the "current state" and returning the `initialState` since the `finishCheckout` function clears the cart items.

I tend to leave the `state` parameter in store handler functions to show the transformation that is happening.

voronianski commented:
@jordangarcia thanks for example :+1: could you please fix code style issues? (https://github.com/voronianski/flux-comparison/blob/master/CONTRIBUTING.md#coding-style, https://github.com/voronianski/flux-comparison/blob/master/.eslintrc)

jordangarcia commented:
Will address all eslint issue

jordangarcia committed:
Fix lint errors

@voronianski I did not fix 'no-use-before-define' errors as its
idiomatic in Nuclear to write the store definition at the top and have
the action handler functions defined below

jordangarcia committed:
Merge remote-tracking branch 'voronianski/master' into nuclear-js

jordangarcia commented:
This should be good now.  Let me know if you'd like me to do anything else before merging!

voronianski commented on package.json:
```diff
@@ -27,6 +27,7 @@
     "object-assign": "^2.0.0",
     "react": "^0.13.0",
     "react-router": "^0.13.0",
+    "nuclear-js": "^1.0.5"
```
@jordangarcia coma is missed

jordangarcia committed:
Fix comma in package.json

jordangarcia commented:
fixed

voronianski commented:
@jordangarcia thanks a lot, I'll check and then merge it.
